backends-common/rabbitmq/src/test/java/org/apache/james/backends/rabbitmq/ChannelPoolContract.java

```diff
@@ -85,8 +84,8 @@ default void channelPoolShouldThrowAfterTimeoutWhenAllChannelsAreTaken() {
         borrowChannel(channelPool);
 
         assertThatThrownBy(channelMono::block)
-            .isInstanceOf(NoSuchElementException.class)
-            .hasMessage("Timeout waiting for idle object");
+            .isInstanceOf(IllegalStateException.class)
+            .hasStackTraceContaining("Timeout waiting for idle object");
```

Comment:
why? I guess it's because you use a checked exception, isn't it?

Reply:
> why?

Because it now is reactor retries exhausted exception. It wraps the actual exception.

Reply:
```java
    static class MyException extends RuntimeException {
    }

    @Test
    void testRetry() {
        Assertions.assertThatThrownBy(() -> Mono.error(MyException::new).retry(2).block()).isInstanceOf(MyException.class);
    }
```
is green

Reply:
Cool, thanks! I didn't got your point before.

Reply:
But wait, did you try with retryBackOff() ?
```java
    static class MyException extends RuntimeException {
    }

    @Test
    void testRetry() {
        assertThatThrownBy(() -> Mono.error(MyException::new)
            .retryBackoff(3, Duration.ofMillis(50), Duration.ofSeconds(1), Schedulers.elastic())
            .subscribeOn(Schedulers.elastic())
            .block())
            .isInstanceOf(MyException.class);
    }
```

Reply:
> But wait, did you try with retryBackOff() ?

No but I guess you did and it's not ok.
What about using `onErrorMap` to workaround that?